bgrep.c

```diff
@@ -177,21 +180,30 @@ void searchfile(const char *filename, int fd, const unsigned char *value, const
 			}
 		}
 
-		offset += r;
-
+		offset += o;
 	}
 
 	free(buf);
 }
 
-void recurse(const char *path, const unsigned char *value, const unsigned char *mask, int len)
+typedef struct Options {
+	char* pattern_path;
+	char* mask_path;
+	bool recurse;
+} Options;
+
+void recurse(const char *path, const unsigned char *value, const unsigned char *mask, int len, const Options* options)
 {
 	struct stat s;
 	if (stat(path, &s))
 	{
 		perror("stat");
 		return;
 	}
+	if (!options->recurse && S_ISDIR(s.st_mode)) {
+		errno = EISDIR;
+		err(1, "%s", path);
```

Comment:
where's `err` defined? it's not available at least on my glibc system.

Reply:
`err(3)` comes from `err.h` and should be part of `libc`. I can check in a bit, but it’s possible that it was generating a warning about a missing declaration I missed.